src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.ConcurrentAccessDetection.netcoreapp.cs

```diff
@@ -0,0 +1,128 @@
+                //Remove is not resilient yet
+                //Assert.Equal("ThrowInvalidOperationException_ConcurrentOperationsNotSupported", Assert.Throws<InvalidOperationException>(() => remove(dictionary)).TargetSite.Name);
+                //Assert.Equal("ThrowInvalidOperationException_ConcurrentOperationsNotSupported", Assert.Throws<InvalidOperationException>(() => removeOutParam(dictionary)).TargetSite.Name);
+            }, TaskCreationOptions.LongRunning);
+
+            //Wait max 60 seconds, could loop forever
+            Assert.True((await Task.WhenAny(task, Task.Delay(TimeSpan.FromSeconds(60))) == task) && task.IsCompletedSuccessfully);
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData(typeof(CustomEqualityComparerInt32ValueType))]
+        async public static Task Add_DictionaryConcurrentAccessDetection_ValueTypeKey(Type comparerType)
+        {
+            IEqualityComparer<int> customComparer = null;
+
+            Dictionary<int, int> dic =
+                comparerType == null ?
+                new Dictionary<int, int>() :
+                new Dictionary<int, int>((customComparer = (IEqualityComparer<int>)Activator.CreateInstance(comparerType)));
+
+            dic.Add(1, 1);
+
+            await DictionaryConcurrentAccessDetection(dic,
+                typeof(int).IsValueType,
+                customComparer,
+                d => d.Add(1, 1),
+                d => { var v = d[1]; },
+                d => d.Remove(1),
+                d => d.Remove(1, out int value));
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData(typeof(CustomEqualityComparerDummyRefType))]
+        async public static Task Add_DictionaryConcurrentAccessDetection_ReferenceTypeKey(Type comparerType)
+        {
+            IEqualityComparer<DummyRefType> customComparer = null;
+
+            Dictionary<DummyRefType, DummyRefType> dic =
+                comparerType == null ?
+                new Dictionary<DummyRefType, DummyRefType>() :
+                new Dictionary<DummyRefType, DummyRefType>((customComparer = (IEqualityComparer<DummyRefType>)Activator.CreateInstance(comparerType)));
+
+            dic.Add(new DummyRefType() { Value = 1 }, new DummyRefType() { Value = 1 });
+
+            await DictionaryConcurrentAccessDetection(dic,
+                typeof(DummyRefType).IsValueType,
+                customComparer,
+                d => d.Add(new DummyRefType() { Value = 1 }, new DummyRefType() { Value = 1 }),
+                d => { var v = d[new DummyRefType() { Value = 1 }]; },
+                d => d.Remove(new DummyRefType() { Value = 1 }),
+                d => d.Remove(new DummyRefType() { Value = 1 }, out DummyRefType value));
+        }
+    }
+}
+
+public class DummyRefType
+{
+    public int Value { get; set; }
```

Comment:
nit: separate methods and properties by a an empty line.

Reply:
done